### PR TITLE
Trim Useless Forms of Materials

### PR DESCRIPTION
**Pull request by bruberu**
This PR removes all material flags on materials that do not have a use in any important crafting recipes. Furthermore, groups of material flags in Materials have been reduced in size in order to make this more organized.

### CODE REVIEW

File: src/main/java/gregtech/loaders/recipe/handlers/PartsRecipeHandler.java [hunk 1]
```diff
@@ -434,6 +433,9 @@ public static void processTurbine(OrePrefix toolPrefix, Material material, Ingot
         //noinspection ConstantConditions
         TurbineRotorBehavior.getInstanceFor(rotorStack).setPartMaterial(rotorStack, material);
 
+        if (material.hasProperty(PropertyKey.GEM))
```
Comment (TechLord22) [inline src/main/java/gregtech/loaders/recipe/handlers/PartsRecipeHandler.java:436]: For some strange reason, Diamond turbines are still registered.

**Review by TechLord22 — CHANGES_REQUESTED** [review]

File: src/main/java/gregtech/api/recipes/RecipeBuilder.java [hunk 2]
```diff
@@ -12,6 +12,7 @@
 import gregtech.api.util.GTLog;
 import gregtech.api.util.GTUtility;
 import gregtech.api.util.ValidationResult;
+import gregtech.common.ConfigHolder;
```
Comment (ALongStringOfNumbers) [inline src/main/java/gregtech/api/recipes/RecipeBuilder.java:15]: Not needed import

File: src/main/java/gregtech/api/recipes/CountableIngredient.java [hunk 3]
```diff
@@ -1,10 +1,14 @@
 package gregtech.api.recipes;
 
+import gregtech.api.unification.OreDictUnifier;
```
Comment (ALongStringOfNumbers) [inline src/main/java/gregtech/api/recipes/CountableIngredient.java:3]: not needed import

File: src/main/java/gregtech/loaders/recipe/handlers/PartsRecipeHandler.java (hunk 1 above)
Comment (ALongStringOfNumbers) [inline src/main/java/gregtech/loaders/recipe/handlers/PartsRecipeHandler.java:436]: This should not be needed any more after my recent commit which blocks gem Turbine Blades

**Review by TechLord22 — APPROVED** [review]

**Pull request merged**